code/game/atoms_health.dm

```diff
@@ -0,0 +1,293 @@
+	return !!health_max
+
+/**
+ * Retrieves the atom's current health, or `null` if not using health
+ */
+/atom/proc/get_current_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_current
+
+/**
+ * Retrieves the atom's maximum health, or `null` if not using health.
+ */
+/atom/proc/get_max_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_max
+
+/**
+ * Retrieves the atom's current damage, or `null` if not using health.
+ * If `use_raw_values` is `TRUE`, uses the raw var values instead of the `get_*` proc results.
+ */
+/atom/proc/get_damage_value(use_raw_values)
+	if (!has_health())
+		return
+	if (use_raw_values)
+		return health_max - health_current
+	else
+		return get_max_health() - get_current_health()
+
+/**
+ * Retrieves the atom's current damage as a percentage where `100%` is `1.00`.
+ * If `use_raw_values` is `TRUE`, uses the raw var values instead of the `get_*` proc results.
+ */
+/atom/proc/get_damage_percentage(use_raw_values)
+	if (!has_health())
+		return
+	var/max_health = use_raw_values ? health_max : get_max_health()
+	return round(get_damage_value(use_raw_values) / max_health, 0.01)
+
+/**
+ * Checks if the atom's health can be restored.
+ * Should be called before `restore_health()` in most cases.
+ * Returns `null` if health is not in use.
+ * NOTE: Does not include a check for death state by default, to allow repairing/healing atoms back to life.
+ */
+/atom/proc/can_restore_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (!damage)
+		return FALSE
+	if (get_current_health() == get_max_health())
+		return FALSE
+	return TRUE
+
+/**
+ * Checks if the atom's health can be damaged.
+ * Should be called before `damage_health()` in most cases.
+ * Returns `null` if health is not in use.
+ */
+/atom/proc/can_damage_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (!is_alive())
+		return FALSE
+	if (!damage || damage < health_min_damage)
+		return FALSE
+	if (get_damage_resistance(damage_type) == 0)
+		return FALSE
+	return TRUE
+
+/**
+ * Checks if the atom is 'alive' or 'dead'.
+ * Returns `null` if health is not in use.
+ */
+/atom/proc/is_alive()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return get_current_health() > 0
+
+/**
+ * Health modification for the health system. Applies `health_mod` directly to `simple_health` via addition and calls `handle_death_change` as needed.
+ * Has no pre-modification checks, you should be using `damage_health()` or `restore_health()` instead of this.
+ * Returns `TRUE` if the death state changes, `null` if the atom is not using health, `FALSE` otherwise.
+ */
+/atom/proc/mod_health(health_mod, damage_type)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	var/death_state = is_alive()
+	health_current = Clamp(health_current + health_mod, 0, get_max_health())
+	post_health_change(health_mod, damage_type)
```

Comment:
I think we discussed this previously - can't blobs just override mod_health()? They are the only thing that actually uses this proc.

Reply:
This is to allow any effects from the health changes to occur before the rest of the mod_health proc runs with the death state changing - I was having issues with atoms, such as blobs, that qdel() themselves on death when I was just straight overriding mod_health itself for things like icon state checks and changes.